apps/freeablo/fagui/guimanager.cpp

```diff
@@ -38,6 +52,116 @@ namespace FAGui
         paused = false;
         showIngameGui();
     }
+    
+    boost::python::list getHotkeyNames()
+    {
+        boost::python::list hotkeynames;
+        
+        hotkeynames.append(Input::getHotkeyName(quit_key));
+        hotkeynames.append(Input::getHotkeyName(noclip_key));
+        hotkeynames.append(Input::getHotkeyName(changelvlup_key));
+        hotkeynames.append(Input::getHotkeyName(changelvldwn_key));
+        return hotkeynames;
+    }
+    
+    boost::python::list getHotkeys()
+    {
+        boost::python::list hotkeys;
+        boost::python::list quit;
+        boost::python::list noclip;
+        boost::python::list changelvlup;
+        boost::python::list changelvldwn;
+        
+        quit.append(Input::convertAsciiToRocketKey(quit_key[0]));
+        noclip.append(Input::convertAsciiToRocketKey(noclip_key[0]));
+        changelvlup.append(Input::convertAsciiToRocketKey(changelvlup_key[0]));
+        changelvldwn.append(Input::convertAsciiToRocketKey(changelvldwn_key[0]));
+        
+        for (int i=1; i<5; i++)
+        {
+            
+            quit.append(quit_key[i]);
+            noclip.append(noclip_key[i]);
+            changelvlup.append(changelvlup_key[i]);
+            changelvldwn.append(changelvldwn_key[i]);
+        }
+
+        hotkeys.append(quit);
+        hotkeys.append(noclip);
+        hotkeys.append(changelvlup);
+        hotkeys.append(changelvldwn);
+        
+        return hotkeys;
+    }
+    
+    void setHotkey(boost::python::list hotkey)
+    {
+        std::string function = boost::python::extract<std::string>(hotkey[0]);
+        boost::python::list pykeys = boost::python::extract<boost::python::list>(hotkey[1]);
+        int keys [5];
+        
+        boost::python::ssize_t n = boost::python::len(pykeys);
+        for(boost::python::ssize_t i=0;i<n;i++) 
+        {
+            keys[i] = boost::python::extract<int>(pykeys[i]);
+            //std::cout  << keys[i] << std::endl;
+        }
+        keys[0] = Input::convertRocketKeyToAscii(keys[0]);
+        //std::cout  << keys[0] << std::endl;
+         
+        if (function == "quit")
```

Comment:
could these if blocks for each key be factored out into a function?
Otherwise there will be an awful lot of copy paste as more keys are added.